File: 2021_09_29/C++/3.cpp
#include <iostream>

int digit_sum(int n) {
	int ans = 0;
	while (n > 0) {
		ans += n % 10;
		n /= 10;
	}
	return ans;
}

bool isPrime(int n) {
	for (int i = 2; i * i <= n; i++)
		if (n % i == 0)
			return false;
	return true;
}

int main() {
	int ans = 0;
	for (int n = 10000; n < 100000; n++)
		if (isPrime(n) && isPrime(digit_sum(n)))
			ans++;
	std::cout << ans << std::endl;
	return 0;
}

File: 2021_09_29/Pascal/3.pas
var ans : longint;
	n : longint;

function digit_sum(n : longint) : longint;
Begin
	digit_sum := 0;
	while (n > 0) do
	begin
		Inc(digit_sum, n mod 10);
		n := n div 10;
	end;
End;

function isPrime(n : longint) : Boolean;
var i : longint;
Begin
	for i := 2 to round(sqrt(n)) do
		if (n mod i = 0) then
		begin
			isPrime := False;
			Exit;
		end;
	isPrime := True;
End;

BEGIN
	ans := 0;
	for n := 10000 to 99999 do
		if isPrime(n) and isPrime(digit_sum(n)) then
			Inc(ans);
	Writeln(ans);
END.
